LibreHardwareMonitorLib/Hardware/Memory/MemoryGroup.cs

```diff
@@ -5,28 +5,137 @@
+        _hardware.Add(new VirtualMemory(settings));
+        _hardware.Add(new TotalMemory(settings));
+
+        //No RAM detected
+        if (!DetectThermalSensors(out List<SPDAccessor> accessors))
+        {
+            //Retry a couple of times
+            //SMBus might not be detected right after boot
+            _timer = new Timer(RetryTime);
+
+            _timer.Elapsed += (_, _) =>
+            {
+                if (_elapsedCounter++ >= RetryCount || DetectThermalSensors(out accessors))
+                {
+                    _timer.Stop();
+                    _timer = null;
+
+                    if (accessors != null)
+                        AddDimms(accessors, settings);
+                }
+            };
+
+            _timer.Start();
+        }
+        else
+        {
+            AddDimms(accessors, settings);
+        }
     }
 
+    public IReadOnlyList<IHardware> Hardware => _hardware;
+
     public string GetReport()
     {
         return null;
     }
 
-    public IReadOnlyList<IHardware> Hardware => _hardware;
-
     public void Close()
     {
         foreach (Hardware ram in _hardware)
             ram.Close();
     }
+
+    private static bool DetectThermalSensors(out List<SPDAccessor> accessors)
+    {
+        lock (_lock)
+        {
+            var list = new List<SPDAccessor>();
+
+            bool ramDetected = false;
+
+            SMBusManager.DetectSMBuses();
+
+            //Go through detected SMBuses
+            foreach (var smbus in SMBusManager.RegisteredSMBuses)
+            {
+                //Go through possible RAM slots
+                for (byte i = SPDConstants.SPD_BEGIN; i <= SPDConstants.SPD_END; ++i)
+                {
+                    //Detect type of RAM, if available
+                    var detector = new SPDDetector(smbus, i);
+
+                    //RAM available and detected
+                    if (detector.Accessor != null)
+                    {
+                        //We are only interested in modules with thermal sensor
+                        if (detector.Accessor is IThermalSensor { HasThermalSensor: true })
+                            list.Add(detector.Accessor);
+
+                        ramDetected = true;
+                    }
+                }
+            }
+
+            accessors = list.Count > 0 ? list : [];
+            return ramDetected;
+        }
+    }
+
+    private void AddDimms(List<SPDAccessor> accessors, ISettings settings)
+    {
+        foreach (var ram in accessors)
+        {
+            //Default value
+            string name = $"DIMM #{ram.Index}";
+
+            //Check if we can switch to the correct page
+            if (ram.ChangePage(PageData.ModulePartNumber))
+            {
+                name = $"{ram.GetModuleManufacturerString()} - {ram.ModulePartNumber()} (#{ram.Index})";
+            }
+
+            var memory = new DimmMemory(ram, name, new Identifier("ram"), settings);
```

Comment:
Identifier should be unique.  If there are multiple accessors, this will generate duplicate identifiers.

Reply:
I will shortly, also will remove the timer.

Reply:
https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/1797